From 52f05965412a7e847b32ac773a46f20ebf5d34a1 Mon Sep 17 00:00:00 2001
From: Petri Virkkunen <petri.virkkunen@qt.io>
Date: Tue, 29 Jul 2025 10:45:31 +0300
Subject: [PATCH 033/239] QQ4A: Reduce amount of spam calls in
 tst_signallistener

To avoid filling the JNI global reference table limit on the CI,
reduce the amount of signals emitted during the tst_signallistener
spam call testcases.

Fixes: QTBUG-138104
Pick-to: 6.9
Change-Id: Iaeefcbef6447586578f2cf2cc8dbfcb7f4536818
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
(cherry picked from commit 21543ca52d4f542b82d62636812af2dae1497196)
---
 .../qtquickview/signallistener/android/qml/TestView.qml    | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/tests/auto/qtquickview/signallistener/android/qml/TestView.qml b/tests/auto/qtquickview/signallistener/android/qml/TestView.qml
index bdc8b06e1f..b5d6916e61 100644
--- a/tests/auto/qtquickview/signallistener/android/qml/TestView.qml
+++ b/tests/auto/qtquickview/signallistener/android/qml/TestView.qml
@@ -148,7 +148,7 @@ Item {
         name: "spamSignaling"
         when: windowShown
 
-        readonly property int spam_count: 100000
+        readonly property int spam_count: 500
 
         function init() { root.wipeSpies() }
 
@@ -158,7 +158,6 @@ Item {
         }
 
         function test_basic() {
-            skip("QTBUG-138104 Global reference table overflow")
             compare(basicSpy.count, 0)
             for (let i = 0; i < spam_count; ++i) {
                 root.basicSignal()
@@ -169,7 +168,6 @@ Item {
         }
 
         function test_int() {
-            skip("QTBUG-138104 Global reference table overflow")
             const testVal = 123
             compare(intSpy.count, 0)
             for (let i = 0; i < spam_count; ++i) {
@@ -184,7 +182,6 @@ Item {
         }
 
         function test_bool() {
-            skip("QTBUG-138104 Global reference table overflow")
             const testVal = false
             compare(boolSpy.count, 0)
             for (let i = 0; i < spam_count; ++i) {
@@ -199,7 +196,6 @@ Item {
         }
 
         function test_double() {
-            skip("QTBUG-138104 Global reference table overflow")
             const testVal = 123.123
             compare(doubleSpy.count, 0)
             for (let i = 0; i < spam_count; ++i) {
@@ -214,7 +210,6 @@ Item {
         }
 
         function test_string() {
-            skip("QTBUG-138104 Global reference table overflow")
             const testVal = "Testing testing, is this thing on?"
             compare(stringSpy.count, 0)
             for (let i = 0; i < spam_count; ++i) {
-- 
2.51.2

